metrics/internal/collectors/object-bucket.go

```diff
@@ -0,0 +1,226 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+
+	rgwadmin "github.com/ceph/go-ceph/rgw/admin"
+	libbucket "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	bktclient "github.com/kube-object-storage/lib-bucket-provisioner/pkg/client/clientset/versioned"
+	"github.com/openshift/ocs-operator/metrics/internal/options"
+	"github.com/prometheus/client_golang/prometheus"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	cephv1listers "github.com/rook/rook/pkg/client/listers/ceph.rook.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+const (
+	bucketProvisionerName = "ceph.rook.io-bucket"
+	rgwServiceName        = "rook-ceph-rgw"
+	svcDNSSuffix          = "svc"
+	prometheusUserName    = "prometheus-user"
+	endPoint              = "Endpoint"
+	accessKey             = "AccessKey"
+	secretKey             = "SecretKey"
+	cephUser              = "cephUser"
+)
+
+var _ prometheus.Collector = &OBCollector{}
+
+// OBCollector is a custom collector for CephObjectStore Custom Resource
+type OBCollector struct {
+	OBSizeTotal           *prometheus.Desc
+	OBSizeMax             *prometheus.Desc
+	OBObjectCountTotal    *prometheus.Desc
+	OBObjectCountMax      *prometheus.Desc
+	ObjectBucketClaimInfo *prometheus.Desc
+	Informer              cache.SharedIndexInformer
+	AllowedNamespaces     []string
+	bktclient             bktclient.Interface
+	rookclient            rookclient.Interface
+	k8sclient             kubernetes.Interface
+}
+
+// NewOBCollector constructs a collector
+func NewOBCollector(opts *options.Options) *OBCollector {
+
+	sharedIndexInformer := CephObjectStoreInformer(opts)
+
+	return &OBCollector{
+		OBSizeTotal: prometheus.NewDesc(
+			"objectbucket_used_bytes",
+			`Total size in bytes consumed on OB`,
+			[]string{"objectbucket", "object_store"},
+			nil,
+		),
+		OBSizeMax: prometheus.NewDesc(
+			"objectbucket_used_max_bytes",
+			`Maximum size in bytes, that can be consumed on OB`,
```

Comment:
This var name and description are not very clear. What exactly does this represent?